### PR TITLE
GDCPE-383: add datadog apm

### PR DESCRIPTION
**Pull request by ProfOak**
- Datadog's ddtrace package works with python2.7 but it pins a bunch of packages
  to versions that don't work in both py27 and py38. Some manual pinning
  had to be done in setup.py to accommodate this.
- Remove test-requirements.txt and references to it. We don't use that
  file in our tox tests.
- Remove pre-commit and detect-secrets from dev dependencies. They don't
  need to be there to run the tests, and were causing the most issues
  around making sub-dependencies work for py27 and py38.

### CODE REVIEW

File: indexd/app.py [hunk 1]
```diff
@@ -11,9 +11,12 @@
 import sys
 import cdislogging
 
+import ddtrace
+
 
 def app_init(app, settings=None):
     app.logger.addHandler(cdislogging.get_stream_handler())
+    ddtrace.patch_all()
```
Comment (mpsolano) [inline indexd/app.py:19]: Is this all that's required to inject data dog tracing into the existing log statements?

**Review by mpsolano — APPROVED** [review]
Looks good!

**Comment by jiakf** [comment]
Had a question: looking at how datadog apm was added to gdcapi https://github.com/NCI-GDC/gdcapi/pull/1536/files, they had to disable all the dd_traces by default in the .travis.yaml and docker-compose-ci.yaml, and then manually enable elasticsearch, sqlalchemy, and flask in the patch call. Did we not have to do that here since the only component indexd is using that we want to trace sqlalchemy?

File: test-requirements.txt [hunk 2]
```diff
@@ -1,6 +0,0 @@
-mock
```
Comment (stilesj-uchicago) [inline test-requirements.txt:1]: That's right. Get ride of test-requirements.txt because it should be requirements.txt for natural ordering!

**Review by stilesj-uchicago — APPROVED** [review]
test-requirements.txt, TestDockerfile! What is the world coming to. DockerfileTest and requirements-test.txt is obviously the correct naming scheme.

**Comment by SalSikandar** [comment]
> Had a question: looking at how datadog apm was added to gdcapi https://github.com/NCI-GDC/gdcapi/pull/1536/files, they had to disable all the dd_traces by default in the .travis.yaml and docker-compose-ci.yaml, and then manually enable elasticsearch, sqlalchemy, and flask in the patch call. Did we not have to do that here since the only component indexd is using that we want to trace sqlalchemy?

We had no issues in indexd. With gdcapi, I found out that the only integration that necessitated the disabling of ddtrace from travis and jenkins was the psycopg integration.

https://ddtrace.readthedocs.io/en/stable/integrations.html#module-ddtrace.contrib.psycopg

I turned it off from GDCAPI for now.

File: indexd/app.py (hunk 1 above)
Comment (SalSikandar) [reply indexd/app.py:19]: https://ddtrace.readthedocs.io/en/stable/api.html#ddtrace.patch_all

> Automatically patches all available modules.

**Pull request merged**